WireDomain/Sources/WireDomain/Repositories/ConversationsLabels/ConversationLabelsLocalStore.swift

```diff
@@ -0,0 +1,154 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import WireDataModel
+
+// sourcery: AutoMockable
+public protocol ConversationLabelsLocalStoreProtocol {
+
+    /// Save label and related conversations objects to local storage.
+    /// - Parameter conversationLabel: conversation label from WireAPI
+
+    func storeLabel(
+        _ conversationLabel: ConversationLabelInfo
+    ) async throws
+
+    /// Delete old `folder` labels and related conversations objects from local storage.
+    /// - Parameter excludedLabels: remote labels that should be excluded from deletion.
+    /// - Only old labels of type `folder` are deleted, `favorite` labels always remain in the local storage.
+
+    func deleteOldLabelsLocally(
+        excludedLabels: [ConversationLabelInfo]
+    ) async throws
+}
+
+public final class ConversationLabelsLocalStore: ConversationLabelsLocalStoreProtocol {
+
+    // MARK: - Error
+
+    enum Error: Swift.Error {
+        case failedToStoreLabelLocally(UUID)
+    }
```

Comment:
Prefer to call the nested error `Failure`.

Reply:
fixed in https://github.com/wireapp/wire-ios/pull/2141/commits/ff83af996921a21f1496f91da4889d35e7168186